nucypher/blockchain/eth/sol/source/contracts/Issuer.sol

```diff
@@ -128,50 +128,40 @@ abstract contract Issuer is Upgradeable {
     )
         internal returns (uint256 amount)
     {
-        if (currentSupply1 == totalSupply || currentSupply2 == totalSupply) {
+        if (currentPeriodSupply == totalSupply) {
             return 0;
         }
 
-        uint256 maxReward = getReservedReward();
-        uint256 currentReward = _currentPeriod <= currentMintingPeriod ?
-            totalSupply - Math.min(currentSupply1, currentSupply2) : maxReward;
+        if (_currentPeriod > currentMintingPeriod) {
+            previousPeriodSupply = currentPeriodSupply;
+            currentMintingPeriod = _currentPeriod;
+        }
+        uint128 currentReward = totalSupply - previousPeriodSupply;
 
         //(totalSupply - currentSupply) * lockedValue * (k1 + allLockedPeriods) / (totalLockedValue * k2)
         uint256 allLockedPeriods =
             AdditionalMath.min16(_allLockedPeriods, rewardedPeriods) + lockedPeriodsCoefficient;
-        amount = (currentReward * _lockedValue * allLockedPeriods) /
+        amount = (uint256(currentReward) * _lockedValue * allLockedPeriods) /
             (_totalLockedValue * miningCoefficient);
 
         // rounding the last reward
+        uint256 maxReward = getReservedReward();
         if (amount == 0) {
             amount = 1;
         } else if (amount > maxReward) {
             amount = maxReward;
         }
 
-        if (_currentPeriod <= currentMintingPeriod) {
-            if (currentSupply1 > currentSupply2) {
-                currentSupply1 += amount;
-            } else {
-                currentSupply2 += amount;
-            }
-        } else {
-            currentMintingPeriod = _currentPeriod;
-            if (currentSupply1 > currentSupply2) {
-                currentSupply2 = currentSupply1 + amount;
-            } else {
-                currentSupply1 = currentSupply2 + amount;
-            }
-        }
+        currentPeriodSupply += uint128(amount);
     }
 
     /**
     * @notice Return tokens for future minting
     * @param _amount Amount of tokens
     */
     function unMint(uint256 _amount) internal {
-        currentSupply1 -= _amount;
-        currentSupply2 -= _amount;
+        previousPeriodSupply -= uint128(_amount);
+        currentPeriodSupply -= uint128(_amount);
```

Comment:
Yeah - this is a wonderful readability improvement to only occupy a single slot for tracking.

Reply:
Why the type casting here - do you expect another type to be possible?

Reply:
type of `amount` is `uint256`